src/apps/trainings/managers/network_queryset.py

```diff
@@ -39,25 +39,25 @@ def select_most_recent(self, run: Run, for_training_games=False, for_rating_game
 
     def select_high_lower_confidence(self, run: Run, for_training_games=False, for_rating_games=False):
         filtered = self.select_networks_for_run(run,for_training_games=for_training_games,for_rating_games=for_rating_games)
-        return filtered.order_by("-log_gamma_lower_confidence").first()
+        return filtered.order_by("-log_gamma_lower_confidence","?").first()
```

Comment:
what is the impact of the random here ? Won't it annihilate every index optimization if the table is big enough ?

Reply:
Replied in discord chat. Apparently this is a bit complicated, postgres 13 has an "incremental sort" feature that allows it to avoid computing random() on the whole table because random() is only the secondary sort field, not the primary ordering field. So it can sort by the primary field log_gamma_lower_confidence first, and then only use random to break ties after that.

However, it seems like based on the analysis, due to the filter criteria, it seems to be doing something involving the whole table anyways? I don't quite understand why. At least, the random isn't making it worse.